global-script.lua

```diff
@@ -2975,7 +3110,20 @@ function setupPlayerArea(params)
         --Updates the number display
     end
 
-    Wait.time(countItems,1,-1)
+    initialized = true
+    obj.setVar("initialized", true)
+
+    if selected then  -- Have a spirit here.
+        countItems()    -- Update counts immediately.
+        if not timer then   -- Timer doesn't already exist.
+            timer = Wait.time(countItems, 1, -1)
+            obj.setVar("timer", timer)
+        end
+    elseif timer then  -- No spirit, but a running timer.
```

Comment:
how would this occur?

Reply:
If you choose a spirit and then move to a different spot on the table, your original location no longer has a spirit but still has the timer for updating element counts already running.  If that's the case, we kill it.

Reply:
shouldn't that logic live in `swapPlayerAreas` then?

Reply:
swapPlayerAreas does have that logic -- by triggering setupPlayerArea to update everything afterwards (which it'd need to do anyways.)

(Really, setupPlayerArea is more of an updatePlayerArea with one-time-setup built in, but it was previously called setupPlayerArea.  It's one of a handful of things that could probably stand to have better nomenclature.)